src/core/util/byte_stream.h

```diff
@@ -138,11 +143,26 @@ class OutputByteStream {
   /// @param data Data to write
   void WriteUInt32(std::uint32_t data);
 
-  uint8_t* GetPosition() const;
+  /// @brief Writes a 64-bit unsigned integer type into buffer
+  /// @note Converts bytes from host to big-endian order
+  /// @note Increments buffer pointer position after writing data
+  /// @param data Data to write
+  void WriteUInt64(std::uint64_t data);
+
+  /// @brief Get current pointer position of written data
+  std::uint8_t* GetPosition() const;
```

Comment:
Are these supposed to be pointers to mutable data? Would `const std::uint8_t*` be better?